lib/P6doc.pm6

```diff
@@ -1,187 +1,356 @@
+            my $cur = @curs.first: {.prefix eq $file.parent.parent}
+            my $dist-with-prefix = $dist but role :: { has $.repo-prefix = $cur.prefix };
+            take $dist-with-prefix;
+        }
+    }
+}
+
+###
+### NEXT
+###
+
+#| Create a Perl6::Documentable::Registry for the given directory
+sub compose-registry(
+    $topdir,
+    @dirs = ['Type'] ) returns Perl6::Documentable::Registry
+{
+    my $registry = process-pod-collection(
+        cache => False,
+        verbose => False,
+        topdir => $topdir,
+        dirs => @dirs
+    );
+    $registry.compose;
+
+    return $registry
+}
+
+#| Receive a list of paths to pod files and process them, return a list of
+#| Perl6::Documentable objects
+sub process-type-pods(IO::Path @files) returns Array[Perl6::Documentable] is export {
+    my Perl6::Documentable @results;
+
+    for @files.list -> $f {
+        my $documentable = process-pod-source(
+            kind => "type",
+            pod => load($f)[0],
+            filename => $f.Str,
+        );
+        @results.push($documentable);
+    }
+
+    return @results;
+}
+
+#| Search for relevant files in a given directory (recursively, if necessary),
+#| and return a list of the results.
+#| $type-name is the name of the type in the form `Str`, `IO::Spec::Unix` etc..
+#| This assumes that $dir is the base directory for the pod files, example: for
+#| the standard documentation folder 'doc', `$dir` should be `'doc'.IO.add('Type')`.
+sub type-list-files(Str $type-name, $dir) returns Array[IO::Path] is export {
+    my IO::Path @results;
+    my $search-name;
+
+    # TODO: Depth not yet optimal
+    my $query-depth = 0..2;
+
+    my $finder = Path::Finder;
+
+    if $type-name.contains('::') {
+        $query-depth = $type-name.split('::').elems + 1;
+        $search-name = $type-name.split('::').tail;
+
+    } else {
+        $search-name = $type-name;
+    }
+
+    $finder = $finder.depth($query-depth);
+    $finder = $finder.name("{$search-name}.pod6");
+
+    for $finder.in($dir, :file) -> $file {
+        @results.push($file);
+    }
+
+    return @results;
+}
+
+#| Search for a single Routine/Method/Subroutine, e.g. `split`
+# TODO: type-search already profits from pre sieving, routine search does
+# not have any optimization yet!
+sub routine-search(
+    Str $routine,
+    :@topdirs = get-doc-locations() ) returns Array[Perl6::Documentable] is export
+{
+    my Perl6::Documentable @results;
+
+    for @topdirs -> $td {
+        my $registry = compose-registry($td);
+
+        # The result from `.lookup` is containerized, thus we use `.list`
+        for $registry.lookup($routine, :by<name>).list -> $r {
+            @results.append: $r;
+        }
+    }
+
+    return @results
+}
+
+#| Lookup documentation in association with a type, e.g. `Map`, `Map.new`.
+#| `$dir` makes the same assumption as `type-list-files`.
+# TODO: might be cleaner to just give it a list of Perl6::Documentable objects
```

Comment:
By having this regular comment after the POD6, the POD6 is no longer attached to the subroutine's name.